### ISSUE TITLE
MOM6 (with ocean only benchmark)  gets stuck with higher rank count per node

### ISSUE THREAD

**Issue by puneet336** [mom-ocean/MOM6#1527]
Hi,
I am trying to run the MOM6 with ocean only benchmark, the simulation seems to be getting stuck.
Compiler - intel2019u5
state of the directory  where simulation got stuck-
```
ocean_only_benchmark1]$ ls test/
available_diags.000000  input.nml           MOM6              MOM_parameter_doc.all        ocean.stats      time_stamp.out
change.sh               input.nml.bak       MOM_input         MOM_parameter_doc.debugging  ocean.stats.nc   Vertical_coordinate.nc
CPU_stats               input.nml.bk        MOM_memory.h      MOM_parameter_doc.layout     RESTART
diag_table              libnuma.so          MOM_override      MOM_parameter_doc.short      run.sh
env.src                 logfile.000000.out  MOM_override.bak  mon                          run.sh.bak
GOLD_IC.nc              lstopo.info         MOM_override.bk   ocean_geometry.nc            slurm-86603.out

```
contents of MOM_override file - 
```
ocean_only_benchmark1]$ cat test/MOM_override
! Blank file in which we can put "overrides" for parameters
#override NIGLOBAL = 720
#override NJGLOBAL = 360

```
contents if input.nml - 
```
ocean_only_benchmark1]$ cat test/input.nml
 &MOM_input_nml
         output_directory = './',
         input_filename = 'n'
         restart_input_dir = 'INPUT/',
         restart_output_dir = 'RESTART/',
         parameter_filename = 'MOM_input',
                              'MOM_override' /

 &diag_manager_nml
 /

 &fms_nml
        clock_grain='ROUTINE'
        clock_flags='SYNC'
        domains_stack_size = 955296
        stack_size =0 /

 &ocean_solo_nml
        months = 0
        days = 20 /
```

[slurm.out.txt](https://github.com/NOAA-GFDL/MOM6/files/7375437/slurm.out.txt)

with 56 ranks (i have 56 cores per node), the simulation seems to be getting stuck at this state (stdout) - 
```
MOM Date      1/01/01 00:00:00      0: En 5.483091E-01, MaxCFL  0.00000, Mass 7.909719100499E+19, Salt  35.00000000000, Temp   5.06383782258
    Total Energy: 4402CF01460DFD5A  4.3369709224187347E+19
    Total Mass:   7.9097191004994732E+19, Change:   0.0000000000000000E+00 Error:  0.00000E+00 ( 0.0E+00)
    Total Salt:   2.7684016851748152E+18, Change:   0.0000000000000000E+00 Error:  0.00000E+00 ( 0.0E+00)
    Total Heat:   1.5721012388222692E+24, Change:   0.0000000000000000E+00 Error:  0.00000E+00 ( 0.0E+00)
      Total age:   0.0000000000000000E+00 yr kg
```

attaching the stdout herewith. Rest other files are same as present in the source code. The same setup runs to completion with lower number of ranks (28). Do i need to modify some settings in the input file to make the simulation work with higher number of ranks (56)?
Please let me know in case some more information is required from my end on this issue.

**Comment by puneet336** [comment]
Thank you for the reply,
We usually run the applications by fully subscribing all the available cores in the system. We tested and benchmarked other Weather softwares like WRF NEMO and ROMS (even synthetic benchmarks like HPL,HPCG) with same compiler version (intel 2019u5).  So far, I have experienced the application hang issue only with MOM6 application. 

I checked and found out that the application stuck issue shows up when i try to launch more than 50 ranks per node. Will check with mpi debuggers to see if i get something from the trace.

I am not expert in weather science but I have seen that one of the causes may be the input dataset too as sometime back with WRF, we noted that due to particular orography the simulation was getting stuck. Could you suggest any other data set for MOM6 which i can try to check  if this is specific to the dataset?


will repost the issue on MOM6 CESM User forum.

**Comment by marshallward** [comment]
I've just tested this on our machine and I am seeing the same problem.  This case ought to work, so this is probably something we should fix.  I can see several ranks are stuck in the bulkmixedlayer (L436) waiting for something, but cannot yet see what is causing the hang.

I will follow up once I've looked through the logs.

**Comment by marshallward** [comment]
This should be resolved by #1542.

I'd suggest questions about comparative performance be moved to the CESM forum.
